src/main_thread/init/utils/create_media_source.ts

```diff
@@ -54,6 +57,42 @@ export function resetMediaElement(
     }
   }
 }
+/**
+ * Temporarily disables remote playback on a media element by setting the
+ * `disableRemotePlayback` attribute to `true` when using a `ManagedMediaSource`.
+ * The original value of the `disableRemotePlayback` attribute is restored when
+ * the cancellation signal is triggered.
+ *
+ * This is useful when the `ManagedMediaSource` is being used and
+ * the media element needs to ensure that remote playback (e.g., Airplay) is disabled
+ * during the playback session.
+ * @param {HTMLElement} mediaElement - The media element whose `disableRemotePlayback`
+ * attribute will be modified.
+ * @param {CancellationSignal} cancellationSignal - The signal that, when triggered,
+ * restores the `disableRemotePlayback` attribute to its original value.
+ */
+export function disableRemotePlayback(
+  mediaElement: IMediaElement,
+  cancellationSignal: CancellationSignal,
+) {
+  if (isManagedMediaSource && "disableRemotePlayback" in mediaElement) {
```

Comment:
It's a weird combination when I read this to:
  - have a method just called `disableRemotePlayback` that takes a media element and a cancel signal (which I would assume just disable remote playback until the cancel signal emits I guess)
  - have it inside `create_media_source` and have it checking `isManagedMediaSource` (and doing nothing if that isn't the case)

Reply:
Yes but the other way around I would need to repeat the condition 
`if (isManagedMediaSource && "disableRemotePlayback" in mediaElement)` every time `disableRemotePlayback()` is used. I found it to be more factorized like that, do you prefer the other way and repeat the condition?

Reply:
Or maybe I could rename it so it is more clear that it could disable it or not depending of the type of mediaSource